core/net/ip/uip-debug.c

```diff
@@ -74,6 +74,10 @@ uip_debug_ipaddr_print(const uip_ipaddr_t *addr)
 void
 uip_debug_lladdr_print(const uip_lladdr_t *addr)
 {
+  if(addr == NULL) {
+    printf("(NULL LL addr)");
+    return;
+  }
   unsigned int i;
```

Comment:
The variable definition should come before the added code, otherwise some older compilers (for platforms using the IPv4 stack) will fail here.